src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const prototype = Object.assign({}, state);
+  const statesHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(prototype, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete prototype[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in prototype) {
+          delete prototype[key];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
you may throw some error here

Reply:
good point :)
As soon as I learn how to handle errors I will :)